### PR TITLE
Test for Shadow bucket in index mode to determine if TAP listener is started

### LINKED ISSUE #2193: Bucket shadowing not working with Distributed Indexing
See https://github.com/couchbaselabs/mobile-testkit/issues/811

### PR DESCRIPTION
**Pull request by ajres**
fixes #2193

### CODE REVIEW

File: db/database.go [hunk 1]
```diff
@@ -194,9 +195,12 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
 	// Initialize the tap Listener for notify handling
 	context.tapListener.Init(bucket.GetName())
 
-	// If not using channel index, start the tap feed
-	if options.IndexOptions == nil {
-		if err = context.tapListener.Start(bucket, options.TrackDocs, func(bucket string, err error) {
+	// If not using channel index or using shadowing, start the tap feed
+	if options.IndexOptions == nil || options.Shadowing {
```
Comment (tleyden) [inline db/database.go:199]: Can this be changed to be more consistent?  Eg `!options.IndexOptions || options.Shadowing` or `options.IndexOptions == nil || options.Shadowing != nil`

File: db/database.go [hunk 2]
```diff
@@ -81,7 +81,8 @@ type DatabaseContextOptions struct {
 	RevisionCacheCapacity uint32
 	AdminInterface        *string
 	UnsupportedOptions    *UnsupportedOptions
-	TrackDocs             bool // Whether doc tracking channel should be created (used for autoImport, shadowing)
+	AutoImport            bool // Whether autoImport of documents without _sync is enabled
```
[the remark this answers is not in the document]
Comment (tleyden) [reply db/database.go:84]: Is `AutoImport` documented more anywhere?  Is `AutoImport` related to bucket shadowing?

**Review by tleyden — CHANGES_REQUESTED** [review]

**Comment by adamcfraser** [comment]
I agree with the comments from @tleyden - I need some context on why TrackDocs is getting changed to AutoImport/Shadowing in this PR.

File: db/database.go (hunk 1 above)
Comment (ajres) [reply db/database.go:199]: I've rebased this change against master and created a new PR which just adds a `Shadowing` property to the `DatabaseContextOptions` struct.

see #2213

**Pull request closed without merging**